sources/packages/backend/apps/api/src/services/education-program/education-program.service.ts

```diff
@@ -803,6 +819,155 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
     });
   }
 
+  /**
+   * Gets the programs query with the query parameters.
+   * @param offeringTypes offering types.
+   * @param institutionId institution id.
+   * @returns programs query.
+   */
+  private getProgramsQueryWithQueryParams(
+    offeringTypes: OfferingTypes[],
+    institutionId: number,
+  ): {
+    programQuery: SelectQueryBuilder<EducationProgram>;
+    queryParams: any[];
+  } {
+    const programQuery = this.repo
+      .createQueryBuilder("programs")
+      .select("programs.id", "programId")
+      .addSelect("programs.name", "programName")
+      .addSelect("programs.cipCode", "cipCode")
+      .addSelect("programs.credentialType", "credentialType")
+      .addSelect("programs.createdAt", "programSubmittedAt")
+      .addSelect("location.id", "locationId")
+      .addSelect("location.name", "locationName")
+      .addSelect("programs.programStatus", "programStatus")
+      .addSelect("programs.isActive", "isActive")
+      .addSelect("programs.effectiveEndDate", "effectiveEndDate")
+      .addSelect(
+        (qb) =>
+          qb
+            .select("COUNT(*)")
+            .from(EducationProgramOffering, "offerings")
+            .where("offerings.educationProgram.id = programs.id")
+            .andWhere("offerings.institutionLocation.id = location.id")
+            .andWhere("offerings.offeringType in (:...offeringTypes)", {
+              offeringTypes,
+            }),
+        "totalOfferings",
+      )
+      .innerJoin("programs.institution", "institution")
+      .innerJoin(
+        InstitutionLocation,
+        "location",
+        "institution.id = location.institution.id",
+      )
+      .where("programs.institution.id = :institutionId", { institutionId });
+
+    // This queryParams is for getRawCount, which is different from the
+    // query parameter assigned to paginatedProgramQuery like
+    // paginationOptions.searchCriteria or institutionId
+    // queryParams should follow the order/index.
+    const queryParams: any[] = [...offeringTypes, institutionId];
```

Comment:
Even being from the old code, it can be changed to `unknown` without a need for further modifications.